interview-questions.md

```diff
@@ -8,62 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: Map iterates through an array and spits out a new array of the same length. Meanwhile filter loops through an array and spits back a new array that is smaller in length. Filter is a smaller length because it is only giving back values that meet the conditions. 
 
-  Researched answer:
+  Researched answer: Filter is a built in method for Javascript. It allows you to filter thorugh the array and create a new filtered array. Map is another javascript built-in method. Maps takes a callback and runs it over each element of the original array, and then it will create a new array based on the existing array adn share its length. Map method is basically used to covert each element in the array, while the filter method selects certain elements of the original array.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: A function can be considered a tiny machine that takes a code and performs a specific set of tasks. A method are actions that are performed on functions. 
 
-  Researched answer:
+  Researched answer: Javascript functions perform tasks using block of code. To execute a function, the funcion would need to be invoked. Functions contain paramters that are listed after the name of the function in parenthesis. The function then takes in an argument that are values. Functions have return statements that will allow the function to stop being executed and return a value. Javascript methods is a property of an object. They are essentially functions stored as object properties. Method will operate the data in a class while function can use the data that is operated and return the data. Function is its own entity while the method is associated with object property. 
 
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer: Object destructuring is a way to assign variables. It will break down an object in order to make it into new variables. 
 
-  Researched answer:
+  Researched answer: Object destructuring allows us to take an array or object and turn it into individual variables. This is a way of assigning variables. And essentially you are taking the properties of an object and breaking them so they are recallable and individualized variables. The way it works is: you satrt with a variable keyword, list the variable names in curly brackets (making sure it connects to the properties in objects), and then uses the assingment operatot to assign the object deconstructed. 
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: I believe hoisting is the ability to use functions and variables before they're executed. 
 
-  Researched answer:
+  Researched answer: Hositing refers to the process where a variable can be used before it has been declared. It moves declarations to the top of their scope.  
```

Comment:
Awesome work in this section!  Answers are well thought out and thorough!  Great job!!!